wandb/data_types.py

```diff
@@ -873,9 +877,47 @@ def __init__(self, data_or_path, **kwargs):
     def get_media_subdir(self):
         return os.path.join("media", "object3D")
 
-    def to_json(self, run):
-        json_dict = super(Object3D, self).to_json(run)
-        json_dict["_type"] = "object3D-file"
+    @classmethod
+    def from_json(cls, json_obj, source_artifact):
+        return cls(source_artifact.get_path(json_obj["path"]).download())
+
+    def to_json(self, run_or_artifact):
+        json_dict = super(Object3D, self).to_json(run_or_artifact)
+        json_dict["_type"] = Object3D.artifact_type
+
+        _, wandb_artifacts = _safe_sdk_import()
+
+        if isinstance(run_or_artifact, wandb_artifacts.Artifact):
```

Comment:
This block is redundant with the code executed in the Media superclass. I think you can remove this entire conditional except for your path type check.

Reply:
got rid of it and confirmed it still works! Is this newly-redundant with recent changes in master? It's all modeled after stuff `Image` does, and IIRC when I first made this branch it didn't work without it.

Reply:
Yeah, this was in the refactor we were discussing in slack thread - since all these are file-based assets, we can pull up the logic to Media - hopefully it made the changes smoother (: